### ISSUE TITLE
Error when i try to prune my repo and after mutiple rebuild-index

### ISSUE THREAD

**Issue by RomyxBaps** [restic/restic#3836]
Hello everyone,

I have an error when i try to prune the backup after somes errors on my script who backup a folder.

Output of `restic version`
--------------------------
restic 0.13.1 compiled with go1.18 on linux/amd64

How did you run restic exactly?
-------------------------------
CLI on my linux serveur and in a script

What backend/server/service did you use to store the repository?
----------------------------------------------------------------
Swift OVH v3

Expected behavior
-----------------

The command run in CLI are working (prune)

Actual behavior
---------------

I have a lot of error from the CLI.
With prune :
```
Load(<snapshot/bc5e89b25b>, 0, 0) returned error, retrying after 10.089915114s: conn.ObjectOpen: Object Not Found
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x20 pc=0xc4dcd4]

goroutine 121 [running]:
main.getUsedBlobs.func1({0xe1, 0xcc, 0x4e, 0xfc, 0xed, 0x84, 0x56, 0xda, 0x0, 0x3c, ...}, ...)
	/restic/cmd/restic/cmd_prune.go:558 +0x54
github.com/restic/restic/internal/restic.ForAllSnapshots.func2()
	/restic/internal/restic/snapshot.go:113 +0x185
golang.org/x/sync/errgroup.(*Group).Go.func1()
	/home/build/go/pkg/mod/golang.org/x/sync@v0.0.0-20210220032951-036812b2e83c/errgroup/errgroup.go:57 +0x67
created by golang.org/x/sync/errgroup.(*Group).Go
	/home/build/go/pkg/mod/golang.org/x/sync@v0.0.0-20210220032951-036812b2e83c/errgroup/errgroup.go:54 +0x8d
```

I'm connect on my swift with cyberduck and i see the snapshot.

Also, when i try to `forget` the snaphots, he is not found : 
```
Load(<snapshot/bc5e89b25b>, 0, 0) returned error, retrying after 15.406612236s: conn.ObjectOpen: Object Not Found
Ignoring "bc5e89b25b7ad312682be376e69d981509eea2d81ef693c318fd4dc4c90bb736", could not load snapshot: conn.ObjectOpen: Object Not Found
```

Steps to reproduce the behavior
-------------------------------

I don't really know, it appear after mutliple backup with errors

Do you have any idea what may have caused this?
-----------------------------------------------

Maybe a connection lost ? Or mutiple errors ?

Do you have an idea how to solve the issue?
-------------------------------------------

I try a lot of `rebuild-index` and `rebuild-index --read-all-packs`

**Comment by RomyxBaps** [comment]
The problem is resolved, i followed an old issue from me #3433 .
I suppress wiht cyberduck the snapshopt directly that was not found and the prune run correctly.

**Comment by MichaelEischer** [comment]
The crash itself has been fixed by https://github.com/restic/restic/pull/3767 .
